notify/sns/sns.go

```diff
@@ -0,0 +1,190 @@
+			Credentials: creds,
+			Endpoint:    aws.String(tmpl(n.conf.APIUrl)),
+		},
+		Profile: n.conf.Sigv4.Profile,
+	})
+
+	if _, err := sess.Config.Credentials.Get(); err != nil {
+		return false, fmt.Errorf("could not get SigV4 credentials: %w", err)
+	}
+
+	if n.conf.Sigv4.RoleARN != "" {
+		sess.Config.Credentials = stscreds.NewCredentials(sess, n.conf.Sigv4.RoleARN)
+	}
+
+	// Max message size for a message in a SNS publish request is 256KB, except for SMS messages where the limit is 1600 characters/runes.
+	messageSize := 256 * 1024
+	client := sns.New(sess, &aws.Config{Credentials: creds})
+	publishInput := &sns.PublishInput{}
+
+	if n.conf.TopicARN != "" {
+		publishInput.SetTopicArn(tmpl(n.conf.TopicARN))
+
+		if n.isFifo == nil {
+			checkFifo, err := checkTopicFifoAttribute(client, n.conf.TopicARN)
+			if err != nil {
+				if e, ok := err.(awserr.RequestFailure); ok {
+					return n.retrier.Check(e.StatusCode(), strings.NewReader(e.Message()))
+				} else {
+					return true, err
+				}
+			}
+			n.isFifo = &checkFifo
+		}
+		if *n.isFifo {
+			// Deduplication key and Message Group ID are only added if it's a FIFO SNS Topic.
+			key, err := notify.ExtractGroupKey(ctx)
+			if err != nil {
+				return false, err
+			}
+			publishInput.SetMessageDeduplicationId(key.Hash())
+			publishInput.SetMessageGroupId(key.Hash())
+		}
+	}
+	if n.conf.PhoneNumber != "" {
+		publishInput.SetPhoneNumber(tmpl(n.conf.PhoneNumber))
+		// If we have an SMS message, we need to truncate to 1600 characters/runes.
+		messageSize = 1600
+	}
+	if n.conf.TargetARN != "" {
+		publishInput.SetTargetArn(tmpl(n.conf.TargetARN))
+
+	}
+
+	messageToSend, isTrunc, err := validateAndTruncateMessage(tmpl(n.conf.Message), messageSize)
+	if err != nil {
+		return false, err
+	}
+	if isTrunc {
+		attributes[tmpl("truncated")] = &sns.MessageAttributeValue{DataType: aws.String("String"), StringValue: aws.String(tmpl("true"))}
+	}
+	publishInput.SetMessage(messageToSend)
+
+	if n.conf.Subject != "" {
+		publishInput.SetSubject(tmpl(n.conf.Subject))
+	}
+
+	publishInput.SetMessageAttributes(attributes)
+
+	publishOutput, err := client.Publish(publishInput)
+	if err != nil {
+		if e, ok := err.(awserr.RequestFailure); ok {
+			return n.retrier.Check(e.StatusCode(), strings.NewReader(e.Message()))
+		} else {
+			return true, err
+		}
+	}
+
+	level.Debug(n.logger).Log("msg", "SNS publish successfully sent", "message_id", publishOutput.MessageId, "sequence number", publishOutput.SequenceNumber)
+
+	return false, nil
+}
+
+func checkTopicFifoAttribute(client *sns.SNS, topicARN string) (bool, error) {
+	topicAttributes, err := client.GetTopicAttributes(&sns.GetTopicAttributesInput{TopicArn: aws.String(topicARN)})
+	if err != nil {
+		return false, err
+	}
+	ta := topicAttributes.Attributes["FifoTopic"]
+	if aws.StringValue(ta) == "true" {
+		return true, nil
+	}
+	return false, nil
+}
+
+func validateAndTruncateMessage(message string, sizeInBytes int) (string, bool, error) {
```

Comment:
[nit] What we take in input is the limit
```suggestion
func validateAndTruncateMessage(message string, maxSizeInBytes int) (string, bool, error) {
```